libraries/exoplayer/src/main/java/androidx/media3/exoplayer/upstream/Loader.java

```diff
@@ -90,6 +90,18 @@ public interface Loadable {
   /** A callback to be notified of {@link Loader} events. */
   public interface Callback<T extends Loadable> {
 
+    /**
+     * Called when a load has started for the first time or through a retry.
+     *
+     * @param loadable The loadable whose load has completed.
+     * @param elapsedRealtimeMs {@link SystemClock#elapsedRealtime} when the load ended.
```

Comment:
If this method is called to indicate the start of a load, why are we passing the time that it ended (surely that hasn't happened yet, since the end comes after the start)

Similar comment for `loadDurationMs` below - since wouldn't you expect this to be basically zero at this point?

Reply:
Apologies I did not update the javadoc when I had copied it from onLoadCompleted. I will update it

Basically, we want this method for retries too. So the loadDurationMs is actually the time from when startLoading was first called to when a load was actually executed.

The elapsedRealtimeMs helps give us timestamps when retries occur as well since this can happen many times per load

Reply:
> The elapsedRealtimeMs helps give us timestamps when retries occur as well since this can happen many times per load

I think this information is available from the `elapsedRealtimeMs` value passed to the `onLoadError` call associated with the retry?

> So the loadDurationMs is actually the time from when startLoading was first called to when a load was actually executed.

It seems that this could also be tracked by an keeping track of the state of each load in your implementation?

Reply:
> I think this information is available from the elapsedRealtimeMs value passed to the onLoadError call associated with the retry?

Hmm, does `elapsedRealtimeMs` on `onLoadError` tells us the time when a retry is actually being executed? From the code, it seems indicate the time when an error was thrown and schedule `MSG_START` with a  possible `retryDelay`. If there is no retryDelay, I think we can use that `elapsedRealtimeMs`, however if there is a delay, then are you suggesting we add that delay manually onto whatever we get for `onLoadError`?

> It seems that this could also be tracked by an keeping track of the state of each load in your implementation?

We might but I think it involves passing the `startTimeMs` returned from `startLoading` to the callback after the `LoadTask` has started. It's a little bit messy but doable. Any reason to not include this as a param given that all other load methods feed the same parameter?